ndn_hydra/repo/handles/query_handle.py

```diff
@@ -57,9 +57,14 @@ def unlisten(self, prefix):
 
     def _on_interest(self, int_name, int_param, _app_param):
         if not int_param.must_be_fresh or not int_param.can_be_prefix:
+            return 
+        if not _app_param:
+            self.logger.info('Query handle: No querytype')
             return
+
         query = self._get_query_from_interest(Name.to_str(int_name))
-        querytype = Component.to_str(Name.from_str(query)[0])
+        querytype = Component.to_str(_app_param)        
```

Comment:
Note, by doing this, if the client puts more than one component in the Name for AppParams (which happens in in the Prefix type), hydra halts.

I would say querytype = Component.to_str( query[0] )

Reply:
Good catch, that makes sense. Fixed in recent commit.